esp32s3-hal/ld/memory.x

```diff
@@ -11,14 +11,27 @@ RESERVE_RTC_SLOW = 0;
 /* define stack size for both cores */
 STACK_SIZE = 8k;
 
-/* Specify main memory areas */
+VECTORS_SIZE = 0x400;
+
+/* Specify main memory areas
+
+ 40370000 <- IRAM/Icache -> 40378000 <- D/IRAM (I) -> 403E0000
+                            3FC88000 <- D/IRAM (D) -> 3FCF0000 <- DRAM/DCache -> 3FD00000
+
+ Startup code uses the IRAM from 0x403B9000 to 0x403E0000, which is not available for static
+ memory, but can only be used after app starts.
+
+ D cache use the memory from high address, so when it's configured to 16K/32K, the region
+ 0x3FCF000 ~ (3FD00000 - DATA_CACHE_SIZE) should be available. This region is not used as
+ static memory, leaving to the heap.
+*/
 MEMORY
 {
-  vectors_seg ( RX )     : ORIGIN = 0x40370000 + RESERVE_DRAM, len =  1k
-  iram_seg ( RX )        : ORIGIN = 0x40370400 + RESERVE_DRAM, len = 328k - 0x400
+  vectors_seg ( RX )     : ORIGIN = 0x40370000 + RESERVE_ICACHES, len =  1k
+  iram_seg ( RX )        : ORIGIN = 0x40370000 + RESERVE_ICACHES + VECTORS_SIZE, len = 328k - VECTORS_SIZE - RESERVE_ICACHES
+  dram_seg ( RW )        : ORIGIN = 0x3FC80000 + RESERVE_ICACHES + VECTORS_SIZE, len = 328k - VECTORS_SIZE - RESERVE_ICACHES
```

Comment:
```suggestion
  dram_seg ( RW )        : ORIGIN = 0x3FC88000 + VECTORS_SIZE, len = 328k - VECTORS_SIZE - RESERVE_ICACHES
```
The start of the `dram_seg` does not depend on the area reserved for the Instruction Cache, since the access to this area (Internal SRAM 0) is exclusive to the instruction bus.

Reply:
good catch